src/shared/api/plugin-memory-api.ts

```
@@ -0,0 +1,76 @@
+
+function validatePluginId(pluginId: string): string {
+  const trimmed = pluginId.trim();
+  if (!PLUGIN_ID_PATTERN.test(trimmed)) {
+    throw new Error("Plugin memory pluginId must be a stable plugin id.");
+  }
+  return trimmed;
+}
+
+function validateMemoryKey(key: string): string {
+  const trimmed = key.trim();
+  if (!trimmed || trimmed.length > MAX_PLUGIN_MEMORY_KEY_LENGTH || /[\u0000-\u001f\u007f]/.test(trimmed)) {
```

ESLint

[error] Unexpected control character(s) in regular expression: \x00, \x1f.

(no-control-regex)